lib/data/repositories/v2/leaderboard_repository.dart

```diff
@@ -27,21 +27,24 @@ class LeaderboardRepository {
   final CoffeecardApiV2 apiV2;
   final Executor executor;
 
-  Future<Either<RequestFailure, List<LeaderboardUser>>> getLeaderboard(
+  Future<Either<ServerFailure, List<LeaderboardUser>>> getLeaderboard(
     LeaderboardFilter category,
   ) async {
-    return executor.execute(
+    final result = await executor(
       () => apiV2.apiV2LeaderboardTopGet(preset: category.label, top: 10),
-      (dtoList) => dtoList.map(LeaderboardUser.fromDTO).toList(),
     );
+
+    return result
+        .bind((result) => Right(result.map(LeaderboardUser.fromDTO).toList()));
   }
 
-  Future<Either<RequestFailure, LeaderboardUser>> getLeaderboardUser(
+  Future<Either<ServerFailure, LeaderboardUser>> getLeaderboardUser(
     LeaderboardFilter category,
   ) async {
-    return executor.execute(
+    final result = await executor(
       () => apiV2.apiV2LeaderboardGet(preset: category.label),
-      LeaderboardUser.fromDTO,
     );
+
+    return result.bind((result) => Right(LeaderboardUser.fromDTO(result)));
```

Comment:
Use map instead of bind
```suggestion
    return result.bind(LeaderboardUser.fromDTO);
```